src/Microsoft.OData.Core/UriParser/SemanticAst/PathSelectItem.cs

```diff
@@ -133,5 +133,24 @@ public override void HandleWith(SelectItemHandler handler)
         {
             handler.Handle(this);
         }
+
+        /// <summary>
+        /// Returns whether or not the path select item has any options applied
+        /// </summary>
+        internal bool HasOptions

```

Comment:
discuss:   consider to public it? It seems useful and better to add a property into the SelectItem (base class). Anyway, i can do it later.